wcfsetup/install/files/ts/WoltLabSuite/Core/Ui/Poll/Editor.ts

```diff
@@ -0,0 +1,404 @@
+    pollOptionInput.appendChild(addButton);
+
+    const deleteButton = document.createElement('a') as HTMLAnchorElement;
+    deleteButton.setAttribute('role', 'button');
+    deleteButton.setAttribute('href', '#');
+    deleteButton.classList.add('icon', 'icon16', 'fa-times', 'jsTooltip', 'jsDeleteOption', 'pointer');
+    deleteButton.setAttribute('title', Language.get('wcf.poll.button.removeOption'));
+    deleteButton.addEventListener('click', (ev) => this.removeOption(ev));
+    pollOptionInput.appendChild(deleteButton);
+
+    // input field
+    const optionInput = document.createElement('input') as HTMLInputElement;
+    optionInput.type = 'text';
+    optionInput.value = optionValue;
+    optionInput.maxLength = 255;
+    optionInput.addEventListener('keydown', (ev) => this.optionInputKeyDown(ev));
+    optionInput.addEventListener('click', () => {
+      // work-around for some weird focus issue on iOS/Android
+      if (document.activeElement !== optionInput) {
+        optionInput.focus();
+      }
+    });
+    pollOptionInput.appendChild(optionInput);
+
+    if (insertAfter !== null) {
+      optionInput.focus();
+    }
+
+    this.optionCount++;
+    if (this.optionCount === this.options.maxOptions) {
+      this.optionList.querySelectorAll('.jsAddOption').forEach((icon: HTMLSpanElement) => {
+        icon.classList.remove('pointer');
+        icon.classList.add('disabled');
+      });
+    }
+  }
+
+  /**
+   * Populates the option list with the current options.
+   */
+  private createOptionList(pollOptions: PollOption[]): void {
+    pollOptions.forEach((option) => {
+      this.createOption(option.optionValue, option.optionID);
+    });
+    
+    if (this.optionCount < this.options.maxOptions) {
+      this.createOption();
+    }
+  }
+
+  /**
+   * Handles validation errors returned by Ajax request.
+   */
+  private handleError(data: AjaxResponse): void {
+    switch (data.returnValues.fieldName) {
+      case this.wysiwygId + 'Poll_endTime':
+      case this.wysiwygId + 'Poll_maxVotes':
+        const fieldName = data.returnValues.fieldName.replace(this.wysiwygId + 'Poll_', '');
+
+        const small = document.createElement('small');
+        small.classList.add('innerError');
+        small.innerHTML = Language.get('wcf.poll.' + fieldName + '.error.' + data.returnValues.errorType);
+
+        const element = document.createElement(data.returnValues.fieldName);
+
+        (element.nextSibling! as HTMLElement).insertAdjacentElement("afterbegin", small);
+
+        data.cancel = true;
+        break;
+    }
+  }
+
+  /**
+   * Adds another option field below the current option field after pressing Enter.
+   */
+  private optionInputKeyDown(event: KeyboardEvent): void {
+    if (event.key !== 'Enter') {
+      return;
+    }
+
+    const target = event.currentTarget as HTMLInputElement;
+    const addOption = target.parentElement!.querySelector('.jsAddOption') as HTMLSpanElement;
```

Comment:
The cast is not helpful here, because we don't care about the type of the element. If just the existence is required a `!` is preferable.